core/src/main/java/org/commonjava/maven/galley/internal/TransferManagerImpl.java

```diff
@@ -626,8 +631,15 @@ public Transfer store( ConcreteResource resource , final InputStream stream , fi
         OutputStream out = null;
         try
         {
+            String content = IOUtils.toString( stream );
```

Comment:
If the content is a 200Mbyte jar this code could be pretty bad for memory performance, since it's storing the whole thing in memory.